src/aggregator/client/queue.go

```diff
@@ -175,10 +182,8 @@ func (q *queue) Enqueue(buf protobuf.Buffer) error {
 		return nil
 	}
 
-	q.mtx.Lock()
-	defer q.mtx.Unlock()
-
-	if full := q.buf.full(); full {
+	full := q.buf.full()
+	for full {
```

Comment:
Hmm so this change probably isn't wrong per se, but it's a bit confusing to me--will this loop ever execute more than once? 

It seems like dropping one item off the queue (in `case DropOldest`) should make the queue no longer full, right? Anything I'm missing there?

Reply:
Yes it can run more than once now because we added a queueSizeBytes limit as opposed to just the number of items before. For instance if we enqueue a bigBuf after many smallBufs then on the next Enqueue() multiple smallBufs would have to be Dequeued() first since the policy is DropOldest.